### PR TITLE
improve performance of loadPlan and loadCurrent

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Improve loadPlan/loadCurrent performance.

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *all cluster tests*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11334/

### CODE REVIEW

**Comment by jsteemann** [comment]
Tests blue except unrelated test failure

File: 3rdParty/velocypack/src/StringRef.cpp [hunk 1]
```diff
@@ -99,9 +103,16 @@ std::size_t StringRef::find(char c) const {
   return (p - _data);
 }
   
-std::size_t StringRef::rfind(char c) const {
+std::size_t StringRef::rfind(char c, std::size_t offset) const noexcept {
+  std::size_t length;
+  if (offset == std::string::npos || offset >= _length + 1) {
```
Comment (graetzer) [inline 3rdParty/velocypack/src/StringRef.cpp:108]:
```suggestion
  if (offset >= _length + 1) {
```
isn't npos `-1` ? Second part should always be true then

File: arangod/Cluster/ClusterInfo.cpp [hunk 2]
```diff
@@ -701,32 +691,39 @@ void ClusterInfo::loadPlan() {
...
+  auto resetLoader = scopeGuard([this, start]() {
     _newPlannedViews.clear();
+    _planLoader = std::thread::id();
+
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    auto diff = clock::now() - start;
+    if (diff > std::chrono::milliseconds(500)) {
+      LOG_TOPIC("66666", WARN, Logger::CLUSTER) << "Loading the new plan took: " << std::chrono::duration<double>(diff).count() << "s";
+    }
+#endif
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp:703]:
```suggestion
#else 
    (void) start;
#endif
```

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11462/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**